pkg/server/image_pull.go

```diff
@@ -247,7 +250,26 @@ func (c *criContainerdService) pullImage(ctx context.Context, ref string) (
 		return "", "", fmt.Errorf("failed to wait for image %q downloading: %v", ref, err)
 	}
 	glog.V(4).Infof("Finished downloading resources for image %q", ref)
+	if schema1Converter != nil {
+		desc, err = schema1Converter.Convert(ctx)
+		if err != nil {
+			return "", "", fmt.Errorf("failed to convert schema 1 image %q: %v", ref, err)
+		}
+	}
 
+	// In the future, containerd will rely on the information in the image store to perform image
+	// garbage collection.
+	// For now, we simply use it to store and retrieve information required for pulling an image.
+	// @stevvooe said we should `Put` before downloading content, However:
+	// 1) Containerd client put image metadata after downloading;
```

Comment:
works for me should probably discuss with stevvooe maybe they client is wrong.. or maybe the advise has changes...